### PR TITLE
integrate hipchat notification

### PR DESCRIPTION
**Pull request by pabin2**
I've integrated hipchat notification in this branch .

### CODE REVIEW

File: test/services/hipchat.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import sinon from 'sinon';
+import faker from 'faker';
+import { assert } from 'chai';
+import rp from 'request-promise';
+import config from '../../src/config/config';
+import * as hipchat from '../../src/services/slack';
```
Comment (bishalshr) [inline test/services/hipchat.test.js]: '../../src/services/slack' ??? :D

Comment (pabin2) [inline test/services/hipchat.test.js]: :ok:

File: test/services/hipchat.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  });
+
+  it('should return true if hipchat notification is enabled.', () => {
+    sandbox.stub(config.notifications, 'hipchat', {
+      enabled: true,
+      roomId: faker.random.word(),
+      authToken : faker.random.word()
+    });
+
+    assert.isTrue(slack.isEnabled());
```
Comment (mesaugat) [inline test/services/hipchat.test.js]: `hipchat.isEnabled()` ??

File: chill.yml.dist [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
   minInterval: 1000
   maxInterval: 10000
+  method: OPTIONS
   downStatus: '^(5..|4..)$'
 
 notifications:
   slack:
     enabled: false
     endpoint: SLACK_WEBHOOK_ENDPOINT_HERE
+  hipchat:
```
Comment (mesaugat) [inline chill.yml.dist]: We probably need to add this to the default config as well. https://github.com/leapfrogtechnology/chill/blob/master/src/config/default.config.js

File: src/services/hipchat.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    logger.info('Sent notification to hipchat.');
+    logger.debug('Result:', result);
+  } catch (err) {
+    logger.error('Error sending notification to hipchat.', err);
+  }
+}
+
+/**
+ * Create and return the payload for the hipchat.
```
Comment (mesaugat) [inline src/services/hipchat.js]: `Create and return the payload for hipchat.`

File: src/services/hipchat.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            title: 'Service'
+          }
+        ]
+      }
+    ]
+  };
+}
+
+/**
+ * Hit the hipchat API endpoint to send notifications on hipchat.
```
Comment (mesaugat) [inline src/services/hipchat.js]: `Hit the hipchat API endpoint to send notifications.`

File: chill.test.yml [hunk 6]
```diff
@@ -11,6 +11,10 @@ notifications:
   slack:
     enabled: false
     endpoint: SLACK_WEBHOOK_ENDPOINT_HERE
+  hipchat:
+    enabled: false
+    authToken: HIPCHAT_WEBHOOK_TOKENID_HERE
```
Comment (mesaugat) [inline chill.test.yml]: We can remove `_HERE`

File: chill.test.yml [hunk 7]
```diff
@@ -11,6 +11,10 @@ notifications:
   slack:
     enabled: false
     endpoint: SLACK_WEBHOOK_ENDPOINT_HERE
```
Comment (mesaugat) [inline chill.test.yml]: same

File: src/services/hipchat.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+ * @returns {Promise}
+ */
+export async function notify(params) {
+  if (!isEnabled()) {
+    return Promise.resolve();
+  }
+
+  logger.debug('Notification Params:', params);
+  let payload = preparePayload(params);
+  try {
```
Comment (mesaugat) [inline src/services/hipchat.js]: Need a new line here. `npm run lint:fix`

File: test/services/hipchat.test.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should return true if hipchat notification is enabled.', () => {
+    sandbox.stub(config.notifications, 'hipchat', {
+      enabled: true,
+      roomId: faker.random.word(),
+      authToken : faker.random.word()
```
Comment (mesaugat) [inline test/services/hipchat.test.js]: `authToken:` - `npm run lint:fix` should fix this as well.

File: src/services/hipchat.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+/**
+ * Hit the hipchat API endpoint to send notifications on hipchat.
+ *
+ * @param {Object} payload
+ * @returns {Promise}
+ */
+function sendNotification(payload) {
+  const url = HOOK_BASE_URI + config.notifications.hipchat.roomId + '/notification?auth_token=' + config.notifications.hipchat.authToken;
```
Comment (kabirbaidhya) [inline src/services/hipchat.js]: How about this?
```javascript
const { roomId, authToken } = config.notifications.hipchat;
const url  = HOOK_BASE_URI + `${roomId}/notification?auth_token=${authToken}`;
```

File: src/services/hipchat.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+ * @returns {Object}
+ */
+function preparePayload(params) {
+  const { status, name } = params;
+
+  let { text, color } = messages[status];
+
+  return {
+    message: text(name, params.downtime),
+    attachments: [
```
Comment (kabirbaidhya) [inline src/services/hipchat.js]: How come the payload for hipchat exactly same as slack's :smile:? https://github.com/leapfrogtechnology/chill/blob/master/src/services/slack.js#L50-L66

Comment (pabin2) [inline src/services/hipchat.js]: I copied all from slack :'(

Comment (kabirbaidhya) [inline src/services/hipchat.js]: But have you tested this? I don't think the payload for hipchat would be same as slack.

Comment (mesaugat) [inline src/services/hipchat.js]: https://www.hipchat.com/docs/apiv2/method/send_room_notification

Comment (badyakarprabin) [inline src/services/hipchat.js]: fixed and pushed :+1:

**Pull request merged**